core/src/components/button/button.ionic.scss

```diff
@@ -0,0 +1,234 @@
+    @include property-horizontal("margin", null, pxToRem(4));
+  }
+  
+  ::slotted(ion-icon[slot=end]) {
+    @include property-horizontal("margin", pxToRem(4), null);
+  }
+}
+
+/* Extra Small Button */
+:host(.button-xsmall) {
+  --border-radius: #{$button-ionic-default-xsmall-border-radius};
+  --padding-top: #{$button-ionic-xsmall-padding-top};
+  --padding-end: #{$button-ionic-xsmall-padding-end};
+
+  min-height: #{$button-ionic-xsmall-min-height};
+  
+  font-size: #{$button-ionic-xsmall-font-size};
+}
+
+/* Small Button */
+:host(.button-small) {
+  --border-radius: #{$button-ionic-default-small-border-radius};
+  --padding-top: #{$button-ionic-small-padding-top};
+  --padding-end: #{$button-ionic-small-padding-end};
+
+  min-height: #{$button-ionic-small-min-height};
+  
+  font-size: #{$button-ionic-small-font-size};
+}
+
+/* Large Button */
+:host(.button-large) {
+  --padding-top: #{$button-ionic-large-padding-top};
+  --padding-end: #{$button-ionic-large-padding-end};
+
+  min-height: #{$button-ionic-large-min-height};
+  
+  font-size: #{$button-ionic-large-font-size};
+}
+
+/* Extra Large Button */
+:host(.button-xlarge) {
+  --padding-top: #{$button-ionic-xlarge-padding-top};
+  --padding-end: #{$button-ionic-xlarge-padding-end};
+
+  min-height: #{$button-ionic-xlarge-min-height};
+
+  font-size: #{$button-ionic-xlarge-font-size};
+}
+
+// Button with Icons
+// -------------------------------------------------------------------------------
+
+/* Button containing only an icon */
+::slotted(ion-icon[slot=start]),
+::slotted(ion-icon[slot=end]),
+::slotted(ion-icon[slot=icon-only]) {
+  font-size: 1em;
+}
+
+/* Button extra small */
+:host(.button-has-icon-only.button-xsmall) {
+  --padding-end: #{$button-has-icon-only-padding-end-xsmall};
+}
+
+/* Button small */
+:host(.button-has-icon-only.button-small) {
+  --padding-end: #{$button-has-icon-only-padding-end-small};
+}
+
+/* Default */
+:host(.button-has-icon-only) {
+  --padding-end: #{$button-has-icon-only-padding-end};
+}
+
+/* Button large */
+:host(.button-has-icon-only.button-large) {
+  --padding-end: #{$button-has-icon-only-padding-end-large};
+}
+
+/* Button extra large */
+:host(.button-has-icon-only.button-xlarge) {
+  --padding-end: #{$button-has-icon-only-padding-end-xlarge};
+}
+
+// Button Shapes
+// -------------------------------------------------------------------------------
+
+/* Rectangular */
+:host(.button-rectangular) {
+  --border-radius: #{$button-ionic-default-rectangular-border};
+}
+
+/* Round */
+:host(.button-round) {
+  --border-radius: #{$button-ionic-default-round-border};
+}
+
+// Button States
+// -------------------------------------------------------------------------------
+
+/* Activated */
+:host(.ion-activated) {
+  --background: #{ion-color(primary, shade)};
+}
+
+:host(.button-outline.ion-activated),
+:host(.button-clear.ion-activated) {
+  --color: #{ion-color(primary, base)};
+  --background: #e3e3e3;
+}
```

Comment:
```suggestion

```

We should actually remove this and set the `--background-activated` and `--color-activated` variables inside of `.button-solid`, `.button-outline` and `.button-clear`. This is my fault for including this in the design doc.